File: project_euler/Problem045/Problem045.js
function Problem045() {
    let index = 144
    while (true) {
        let hexagon = index * ((index << 1) - 1)
        if (isPentagon(hexagon)) {
            console.log(hexagon)
            return
        }
        ++index
    }
}

function isPentagon(value) {
    let auxiliary = Math.sqrt(value * 24 + 1)
    return Math.abs(Math.floor(auxiliary + 0.5) - auxiliary) < 0.00000001 && !((Math.floor(auxiliary) + 1) % 6)
}

module.exports = {
    Problem045 : Problem045
};

File: project_euler/Problem045/Problem045.go
package project_euler

import (
    "fmt"
    "math"
)

func Problem045() {
    index := int64(144)
    for true {
        hexagon := index * ((index << 1) - 1)
        if isPentagon(hexagon) {
            fmt.Println(hexagon)
            return
        }
        index++
    }
}

func isPentagon(value int64) bool {
    auxiliary := math.Sqrt(float64(value * 24 + 1))
    return math.Abs(math.Floor(auxiliary + 0.5) - auxiliary) < 0.00000001 && 0 == (int64(auxiliary) + 1) % 6
}
